c10/util/BFloat16.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+// Defines the bloat16 type (brain floating-point). This representation uses
+// 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
+
+#include <c10/macros/Macros.h>
+#include <cmath>
+#include <cstring>
+
+namespace c10 {
+
+namespace detail {
+  inline bool isSmallEndian() {
+    int num = 1;
+    return *(char *)&num == 1;
+  }
+
+  inline float f32_from_bits(uint16_t src) {
+    float res = 0;
+    uint32_t tmp = src;
+    if (isSmallEndian()) {
+      tmp <<= 16;
+    }
+
+    memcpy(&res, &tmp, sizeof(tmp));
+    return res;
+  }
+
+  inline uint16_t bits_from_f32(float src) {
+    uint32_t res;
+    memcpy(&res, &src, sizeof(res));
+    if (isSmallEndian()) {
+        return res >>= 16;
```

Comment:
I'm still not convinced this is correct for big endian; I don't see anything you are doing here that is different on big endian and small endian (the code looks correct for small endian).